src/schema/organization/contributors/resolvers.js

```diff
@@ -1,61 +1,131 @@
-        const firstContributions = [firstPullRequest, firstIssue, firstReview];
-        const dates = [];
-        
-        firstContributions.forEach(contribution => {
-          if (contribution) dates.push(contribution.firstContributionDate);
-        });
-
-        currentDate = getOldestDate(dates);
-      }
-
-      user.contributionsCollection = undefined;
-      user.firstContributionDate = currentDate;
+      user.a = 1;
       contributors.add(user);
     });
   });
   
   return Array.from(contributors.values());
 };
 
+/**
+ * Process the user object and returns the Date of the first contribution or undefined
+ * if the user has no contributions.
+ * @param {Object} user 
+ * @return {Date} first contribution date
+ */
+const processFirstContributionDate = (user) => {
+  if (user) {
+    const issuesNodes = user.contributionsCollection.issueContributions.nodes;
+    const pullRequestsNodes = user.contributionsCollection.pullRequestContributions.nodes;
+    const reviewsNodes = user.contributionsCollection.pullRequestReviewContributions.nodes;
+    const contributions = [issuesNodes, pullRequestsNodes, reviewsNodes];
+
+    const dates = [];
+    var firstContrib;
+    contributions.forEach(contribution => {
+      if (contribution && contribution.length > 0) {
+        firstContrib = contribution[0].occurredAt;
+        dates.push(new Date(firstContrib)); 
+      }
+    });
+
+    var result = undefined;
+    if (dates.length > 0)
+      result = getOldestDate(dates);
+    return result;
+  }
+}
+
+const firstContributionDateResolver = async(parent, args, { token }) => {
+  const { login } = args;
+  const body = githubQueries.firstContributionDate(login, parent.id);
+  const data = await fetchData(body, token);
+  
+  var result = processFirstContributionDate(data.data.user);
+  return result;
+}
+
+const validateLogin = async (login, orgID, token) => {
+  const body = githubQueries.isContributor(login, orgID);
+  const data = await fetchData(body, token);
+  const { user } = data.data;
+  const isValid = (user && user.contributionsCollection.hasAnyContributions);
+  return isValid;
+}
+
 export default {
   Query: {
     contributors: async (parent, args, { token }) => {
       const { first, last, after, before } = args;
-      const { id: orgID } = parent;
-      
-      // TODO: Fazer outra consulta se totalRepos > 100
+
       const repoArgs = { first: parent.totalRepos }; 
       const userArgs = { first };
 
-      // TODO: Otimizar verificando o número de contribuidores obtidos por iteração em repositorio
-      // caso a quantidade já passe de `first`, paramos de iterar.
       const body = githubQueries.contributors(parent, repoArgs, userArgs);
       const data = await fetchData(body, token);
-
-      let contributorsArray = process(data.data.organization.repositories.nodes);
+      
+      var contributorsArray = processContributors(data.data.organization.repositories.nodes);
```

Comment:
```suggestion
    const contributorsArray = processContributors(data.data.organization.repositories.nodes);
```

What is constant is references, so we can change the properties, in this case you can do 
```
contributorsArray.length = first;
```

Reply:
even if your var is `const`